src/lib/gossip_net/libp2p.ml

```diff
@@ -441,7 +441,12 @@ module Make (Rpc_intf : Coda_base.Rpc_intf.Rpc_interface_intf) :
             (* call itself failed *)
             (* TODO: learn what other exceptions are raised here *)
             let exn = Monitor.extract_exn monitor_exn in
-            match exn with
+            match Error.sexp_of_t (Error.of_exn exn) with
```

Comment:
You can avoid the duplicate `Deferred.return (Or_error.of_exn exn)` expression if you rewrite this code as `(match <exp> with ...); Deferred.return (Or_error.of_exn exn)` (moving the returning expression outside of the match, and having the match expression just be typed as `unit`).